src/ParallelKernel/kernel_language.jl

```diff
@@ -53,11 +53,11 @@ macro sync_threads(args...) check_initialized(); checknoargs(args...); esc(sync_
 
 
 ##
-# NOTE: the optional offset parameter of cuDynamicSharedMem is currently not exposed.
 const SHAREDMEM_DOC = """
-    @sharedMem(T, dims)
+    @sharedMem(T, dims, offset::Integer=0)
 
-Create an array that is *shared* between the threads of a block (i.e. accessible only by the threads of a same block), with element type `T` and size specified by `dims`.
+Create an array that is *shared* between the threads of a block (i.e. accessible only by the threads of a same block), with element type `T` and size specified by `dims`. 
+When multiple shared memory arrays are created within a kernel, then all arrays except for the first one typically need to define the `offset` to the base shared memory pointer in bytes (note that the CPU implementation does not require the `offset` and will simply ignore it when present).
 
 !!! note "Note"
     The amount of shared memory needs to specified when launching the kernel (keyword argument `shmem`).
```

Comment:
The amount of shared memory needs to be specified when launching the kernel (keyword argument `shmem`).